src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+import static core.basesyntax.ColorSupplier.getRandomColor;
+
+public class FigureSupplier {
+    public static Figure getRandomFigure() {
+        int figureNum = (int)(Math.random() * 4);
+        switch (figureNum) {
+            case 0:
+                return new Circle(getRandomColor(), getRandomInt());
```

Comment:
let's create private methods for each figure creation and there explicitly create variables to pass to a constructor